lib/improver/tests/utilities/test_mathematical_operations.py

```diff
@@ -0,0 +1,318 @@
+        chosen direction is positive, the resulting coordinate still
+        increases monotonically in the positive direction."""
+        coord_name = "height"
+        direction = "positive"
+        cube = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).ensure_monotonic_increase_in_chosen_direction(
+                    self.ascending_cube))
+        self.assertIsInstance(cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            cube.coord(coord_name).points, self.ascending_height_points)
+
+    def test_ascending_coordinate_negative(self):
+        """Test that for a monotonically ascending coordinate, where the
+        chosen direction is negative, the resulting coordinate now decreases
+        monotonically in the positive direction."""
+        coord_name = "height"
+        direction = "negative"
+        cube = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).ensure_monotonic_increase_in_chosen_direction(
+                    self.ascending_cube))
+        self.assertIsInstance(cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            cube.coord(coord_name).points, self.descending_height_points)
+
+    def test_descending_coordinate_positive(self):
+        """Test that for a monotonically ascending coordinate, where the
+        chosen direction is positive, the resulting coordinate still
+        increases monotonically in the positive direction."""
+        coord_name = "height"
+        direction = "positive"
+        cube = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).ensure_monotonic_increase_in_chosen_direction(
+                    self.descending_cube))
+        self.assertIsInstance(cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            cube.coord(coord_name).points, self.ascending_height_points)
+
+    def test_descending_coordinate_negative(self):
+        """Test that for a monotonically ascending coordinate, where the
+        chosen direction is negative, the resulting coordinate still decreases
+        monotonically in the positive direction."""
+        coord_name = "height"
+        direction = "negative"
+        cube = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).ensure_monotonic_increase_in_chosen_direction(
+                    self.descending_cube))
+        self.assertIsInstance(cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            cube.coord(coord_name).points, self.descending_height_points)
+
+
+class Test_process(IrisTest):
+
+    """Test the process method."""
+
+    def setUp(self):
+        """Set up the cube."""
+        self.height_points = np.array([5., 10., 20.])
+        cube = set_up_height_cube(self.height_points)[:, 0, :, :, :]
+        data = np.ones(cube.shape)
```

Comment:
Would be clearer (more consistent) to use data[0] = ...

Reply:
Changed.